src/resources/2015/Dinsey.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  availableChoiceOptions,
+  myAdventures,
+  myAscensions,
+  runChoice,
+  visitUrl,
+} from "kolmafia";
+import { getRemainingLiver, have } from "../../lib";
+import { get, getNumber, getString } from "../../property";
+import { $item, $location } from "../../template-string";
+
+const kioskUrl = "place.php?whichplace=airport_stench&action=airport3_kiosk";
+const maintUrl = "place.php?whichplace=airport_stench&action=airport3_tunnels";
+
+export const questPreferenceKioskText = new Map<string, string>([
+  ["questEStSocialJusticeII", "Racism Reduction"],
+  ["questEStSocialJusticeI", "Sexism Reduction"],
+  ["questEStFishTrash", "Waterway Debris Removal"],
+  ["questEStZippityDooDah", "Compulsory Fun"],
+  ["questEStSuperLuber", "Track Maintenance"],
+  ["questEStGiveMeFuel", "Electrical Maintenance"],
+  ["questEStNastyBears", "Bear Removal"],
+  ["questEStWorkWithFood", "Guest Sustenance Assurance"],
+]);
+
+export const questPreferenceLocation = new Map<string, Location>([
+  [
+    "questEStSocialJusticeII",
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`,
+  ],
+  ["questEStSocialJusticeI", $location`Pirates of the Garbage Barges`],
+  ["questEStFishTrash", $location`Pirates of the Garbage Barges`],
+  ["questEStZippityDooDah", $location`The Toxic Teacups`],
+  ["questEStSuperLuber", $location`Barf Mountain`],
+  ["questEStGiveMeFuel", $location`The Toxic Teacups`],
+  [
+    "questEStNastyBears",
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`,
+  ],
+  ["questEStWorkWithFood", $location`Barf Mountain`],
+]);
+
+interface questRequirement {
+  item: Item;
```

Comment:
I'd rather use `Item | null` here tbh

Reply:
For the other two optional parameters, do you prefer string | null, or having them be optional?

Reply:
`undefined` is fine for those. In general I *try* to avoid `$proxy[none]` in libram because its not as JSey as using `null` instead. We could even consider getting mafia to return `null` instead of them!

Reply:
The issue there would be casting it to polymorphic functions on the other end, though I don't know any functions that would take a `$proxy[none]` and do anything meaningful anyway.